src/main/webapp/app/entities/programming-exercise/form/programming-exercise-due-date-select.component.ts

```diff
@@ -0,0 +1,77 @@
+            name="dueDate"
+        ></jhi-date-time-picker>
+        <div id="automatic-submission-after-due-date" class="form-check mt-1" *ngIf="programmingExercise.dueDate && programmingExercise.dueDate.isValid()">
+            <label class="form-check-label" for="field_automaticSubmissionRunAfterDueDate">
+                <input
+                    class="form-check-input"
+                    type="checkbox"
+                    name="automaticSubmissionRunAfterDueDate"
+                    id="field_automaticSubmissionRunAfterDueDate"
+                    [disabled]="!programmingExercise.dueDate"
+                    [ngModel]="!!programmingExercise.automaticSubmissionRunDate"
+                    (ngModelChange)="toggleAutomaticSubmissionRun()"
+                    checked
+                />
+                <span jhiTranslate="artemisApp.programmingExercise.automaticSubmissionRunAfterDueDate.title">Automatic Submission Run After Due Date</span>
+                <fa-icon
+                    icon="question-circle"
+                    class="text-secondary"
+                    placement="top"
+                    ngbTooltip="{{ 'artemisApp.programmingExercise.automaticSubmissionRunAfterDueDate.description' | translate }}"
+                ></fa-icon>
+            </label>
+        </div>
+    `,
+})
+export class ProgrammingExerciseDueDateSelectComponent {
+    @Input() programmingExercise: ProgrammingExercise;
+    @Output() onProgrammingExerciseUpdate = new EventEmitter<ProgrammingExercise>();
+
+    @HostBinding('class') class = 'form-group-narrow flex-grow-1 ml-3';
+    @ViewChild(FormDateTimePickerComponent, { static: false }) dateTimePicker: FormDateTimePickerComponent;
+
+    /**
+     * Set the due date. When the due date is set it needs to be checked if the automatic submission date is also set - this should then be updated, too.
+     * @param dueDate of the programming exercise.
+     */
+    public updateDueDate(dueDate: string) {
+        const updatedDueDate = moment(dueDate).isValid() ? moment(dueDate) : null;
+        const updatedSubmissionRunDate =
+            this.programmingExercise.automaticSubmissionRunDate && updatedDueDate && updatedDueDate.isValid() ? updatedDueDate.clone().add(1, 'hours') : null;
+        const updatedProgrammingExercise = { ...this.programmingExercise, dueDate: updatedDueDate, automaticSubmissionRunDate: updatedSubmissionRunDate };
+        this.onProgrammingExerciseUpdate.emit(updatedProgrammingExercise);
+    }
+
+    /**
+     * We currently don't allow the free setting of the automatic submission run date setting, but set it to one hour after the due date.
+     * The method will return immediately if there is no dueDate set.
+     *
+     * Will emit the updated programming exercise.
+     */
+    public toggleAutomaticSubmissionRun() {
+        if (!this.programmingExercise.dueDate) {
+            return;
+        }
+        const updatedProgrammingExercise = {
```

Comment:
Could you just write:
`        this.programmingExercise.automaticSubmissionRunDate = this.programmingExercise.automaticSubmissionRunDate ? null : this.programmingExercise.dueDate.clone().add(1, 'hours');
`?
Why is it necessary to copy the object?

Reply:
This is as designed: I don't alter `this.programmingExercise` because it is passed down from the parent to this component. Instead I create a new object and emit it.